packages/mobile/src/screens/chat-screen/ChatMessageListItem.tsx

```diff
@@ -64,53 +75,174 @@ const useStyles = makeStyles(({ spacing, palette, typography }) => ({
+
+  if (!reaction || !reaction.reaction || !(reaction.reaction in reactionMap)) {
+    return null
+  }
+  const Reaction = reactionMap[reaction.reaction as ReactionTypes]
+  const reactionPositionStyle = [
+    { top: reactionPosition?.y },
+    isAuthor ? { right: reactionPosition?.x } : { left: reactionPosition?.x }
+  ]
+  return (
+    <Reaction
+      style={[styles.reaction, ...reactionPositionStyle]}
+      key={reaction.user_id}
+      isVisible
+    />
+  )
+}
+
 type ChatMessageListItemProps = {
   message: ChatMessage
   hasTail: boolean
-  unreadCount: number
+  shouldShowReaction?: boolean
+  shouldShowDate?: boolean
+  style?: StyleProp<ViewStyle>
 }
 
-export const ChatMessageListItem = ({
-  message,
-  hasTail,
-  unreadCount
-}: ChatMessageListItemProps) => {
-  const styles = useStyles()
-  const palette = useThemePalette()
+const formatChatReactions = (
+  reactions: ChatMessageReaction[],
+  isAuthor: boolean,
+  reactionPosition: { x: number; y: number }
+) => {
+  // When there are multiple reactions, shift the earlier reactions closer
+  // towards the inside of of the message body.
+  if (reactions.length > 1) {
+    return reactions.map((reaction, index) => {
+      return (
+        <ChatReaction
+          key={index}
+          reaction={reaction}
+          reactionPosition={{
+            ...reactionPosition,
+            // (reactions.length - 1 - index) is an "reverse index" that
+            // increases as index decreases
+            x:
+              reactionPosition.x -
+              (reactions.length - 1 - index) * REACTION_SPACE_BETWEEN
+          }}
+          isAuthor={isAuthor}
+        />
+      )
+    })
+  } else {
+    return (
+      <ChatReaction
+        reaction={reactions[0]}
+        reactionPosition={reactionPosition}
+        isAuthor={isAuthor}
+      />
+    )
+  }
+}
 
-  const userId = useSelector(getUserId)
-  const senderUserId = decodeHashId(message.sender_user_id)
-  const isAuthor = senderUserId === userId
+export const ChatMessageListItem = forwardRef<View, ChatMessageListItemProps>(
+  (props: ChatMessageListItemProps, refProp) => {
+    const {
+      message,
+      hasTail,
+      shouldShowReaction = true,
+      shouldShowDate = true,
+      style: styleProp
+    } = props
+    const styles = useStyles()
+    const palette = useThemePalette()
 
-  return (
-    <>
-      <View style={isAuthor ? styles.rootIsAuthor : styles.rootOtherUser}>
-        <View style={[styles.bubble, isAuthor && styles.isAuthor]}>
-          <Text style={[styles.message, isAuthor && styles.messageIsAuthor]}>
-            {message.message}
-          </Text>
-        </View>
-        {hasTail ? (
-          <>
+    const userId = useSelector(getUserId)
+    const senderUserId = decodeHashId(message.sender_user_id)
+    const isAuthor = senderUserId === userId
+    // Offset of reactions in relation to message body parent.
+    const [reactionPosition, setReactionPosition] = useState<{
```

Comment:
Could this be a ref?

Reply:
nice

Reply:
Dang actually ref is giving me issues where it renders with the old values, gonna have to use state.